### PR TITLE
feat(nix-gc): clean up 0-size drv files in under /nix/store

### PR DESCRIPTION
**Pull request by steveej**
the current implementation isn't clean because of multiple aspects which can be improved subsequently. namely:

* no usage of platform conditional. instead using implicit knowledge about the configs.
* the shared cleanup command is duplicated
* on darwin, the nix-gc command is duplicated from the nix-darwin repo

### CODE REVIEW

File: modules/nixos/shared-darwin.nix [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{ config, lib, ...}: {
+  launchd.daemons.nix-gc.command = lib.mkForce ''
```
Comment (ThetaSinner) [inline modules/nixos/shared-darwin.nix]: Worth a comment to explain why this was added?

```suggestion
  # If the machine runs low on disk space it's possible for derivation files to be created but never get content which results in derivations that can't be removed by the gc. This is a workaround which finds and deletes those problem derivations
  launchd.daemons.nix-gc.command = lib.mkForce ''
```

**Review by ThetaSinner — APPROVED** [review]

File: modules/nixos/shared-darwin.nix (hunk 1 above)
Comment (steveej) [reply modules/nixos/shared-darwin.nix]: because of the duplication we may want to copy this to the linux version too, WDYT?

Comment (steveej) [reply modules/nixos/shared-darwin.nix]: i added it to the commit message

Comment (ThetaSinner) [reply modules/nixos/shared-darwin.nix]: Probably worth adding it to both yes

**Pull request merged**